src/main/java/com/socialmedia/service/UserServiceImpl.java

```diff
@@ -0,0 +1,30 @@
+package com.socialmedia.service;
+
+import com.socialmedia.model.ApplicationUser;
+import com.socialmedia.repository.UserRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.Optional;
+
+@Service
+public class UserServiceImpl implements UserService {
+
+  private UserRepository userRepository;
+
+  @Autowired
+  public UserServiceImpl(UserRepository userRepository) {
+    this.userRepository = userRepository;
+  }
+
+  @Override
+  public Optional<ApplicationUser> getCurrentUser() {
+    // TODO this is a draft method just to get some user returned
+    return Optional.of(userRepository.findAll().get(0));
```

Comment:
Almost done! Just left to fix this todo and return real user. Reference - https://www.baeldung.com/get-user-in-spring-security